core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -347,6 +347,17 @@ class BrokerServer(
 
       shareCoordinator = createShareCoordinator()
 
+      /* setup and configure the persister */
+      persister = if (config.shareGroupConfig.shareGroupPersisterClassName.nonEmpty)
+        CoreUtils.createObject[Persister](config.shareGroupConfig.shareGroupPersisterClassName) else NoOpShareStatePersister.getInstance()
+
+      if (shareCoordinator.isDefined) {
```

Comment:
I wonder whether this initialisation logic should be turned around. The share coordinator is not needed if we are using the NoOpShareStatePersister.